spdm.rflx

```diff
@@ -0,0 +1,1496 @@
+                  and (Registry_ID = Reg_TCG
+                       or Registry_ID = Reg_USB
+                       or Registry_ID = Reg_PCI_SIG
+                       or Registry_ID = Reg_MIPI
+                       or Registry_ID = Reg_CXL
+                       or Registry_ID = Reg_JEDEC)
+            then Vendor_ID_Quad
+               if Vendor_ID_Len = 4
+                  and (Registry_ID = Reg_IANA
+                       or Registry_ID = Reg_HD_Base_T);
+         Vendor_ID_Double : Vendor_ID_Double
+            then Opaque_Error_Data
+               with Size => Message'Size - (Vendor_ID_Double'Last - Error_Code'First + 1);
+         Vendor_ID_Quad : Vendor_ID_Quad
+            then Opaque_Error_Data
+               with Size => Message'Size - (Vendor_ID_Quad'Last - Error_Code'First + 1);
+         Opaque_Error_Data : Opaque;
+      end message;
+
+   -- 10.13 RESPOND_IF_READY request message format
+
+   type Respond_If_Ready_Request is
+      message
+         Original_Request_Code : Request_Code;
+         Token : Token;
+      end message;
+
+   -- 10.16 KEY_EXCHANGE request and KEY_EXCHANGE_RSP response messages
+
+   type Session_ID is mod 2 ** 16;
+
+   type Key_Exchange_Request is
+      message
+         Measurement_Summary_Hash_Type : Measurement_Summary_Hash_Type;
+         Slot : Slot;
+         Req_Session_ID : Session_ID;
+         Reserved : Reserved_16;
+         Random_Data : Opaque
+            with Size => 8 * 32;
+         Exchange_Data : Opaque
+            -- ISSUE: Componolit/RecordFlux-specifications#92
+            -- Size of ExchangeData is determined by the algorithm selected in previous ALGORITHMS
+            -- message
+            with Size => 8 * 96;
+         Opaque_Data_Length_1 : Length;
+         Opaque_Data_Length_2 : Length;
+         Opaque_Data : Opaque
+            with Size => 8 * (Opaque_Data_Length_2 * 256 + Opaque_Data_Length_1);
+      end message;
+
+   type Heartbeat_Period is mod 2**8;
+
+   type Key_Exchange_Response is
+      message
+         Heartbeat_Period : Heartbeat_Period;
+         Param_2 : Reserved_8;
+         Resp_Session_ID : Session_ID;
+         Reserved_1 : Reserved_5;
+         Mutual_Auth_Implicit : Boolean;     -- Bit 2
+         Mutual_Auth_Encapsulated : Boolean; -- Bit 1
+         Mutual_Auth : Boolean               -- Bit 0
+            then Reserved_2
+               if --  Bit 1 and Bit 2 shall only be set if Bit 0 is set
+                  (Mutual_Auth = False
+                   or (Mutual_Auth_Implicit = True
+                       or Mutual_Auth_Encapsulated = True))
+                  --  Bit 1 and Bit 2 shall not be set at the same time
+                  and (Mutual_Auth_Implicit = False
+                       or Mutual_Auth_Encapsulated = False);
+         Reserved_2 : Reserved_4;
+         Slot_ID_Param : Narrow_Slot
+            then Random_Data
+               --  If MutAuthRequested = 0x00 this field shall be set to 0 and ignored by the
```

Comment:
We should add a standard reference here.

Reply:
Added in https://github.com/Componolit/RecordFlux-specifications/pull/93/commits/4b2412a417abf4c2f782dc48e50069dea280daf2.